src/pal/prebuilt/inc/corprof.h

```diff
@@ -464,13 +464,13 @@ typedef void __stdcall __stdcall FunctionTailcall2(
     UINT_PTR clientData,
     COR_PRF_FRAME_INFO func);
 
-typedef void __stdcall __stdcall FunctionEnter3( 
+typedef void STDMETHODCALLTYPE STDMETHODCALLTYPE FunctionEnter3(
```

Comment:
@jkotas doesn't this need to change in corprof.idl?

Do the "WithInfo" functions (e.g., FunctionEnter3WithInfo) also need to change?

What about the version 2 and version 1 function types?

Reply:
@BruceForstall we have been forced to make this change because clang5.0 issues inconsistency between this declaration 
```
typedef void __stdcall __stdcall FunctionEnter3( 
```
and the following using in src/vm/eetoprofinterfaceimpl.cpp
```
            // The casts below are to appease rotor.  cl.exe doesn't need them.
            hr = SetEnterLeaveFunctionHooksForJit(
                (m_pEnter3WithInfo != NULL) ?
                    reinterpret_cast<FunctionEnter3 *>(PROFILECALLBACK(ProfileEnter)) :
                    m_pEnter3,
                (m_pLeave3WithInfo != NULL) ?
                    reinterpret_cast<FunctionLeave3 *>(PROFILECALLBACK(ProfileLeave)) :
                    m_pLeave3,
                (m_pTailcall3WithInfo != NULL) ?
                    reinterpret_cast<FunctionTailcall3 *>(PROFILECALLBACK(ProfileTailcall)) :
                    m_pTailcall3);
```
I don't find any uses of other versions

Reply:
> @jkotas doesn't this need to change in corprof.idl?

Yes, this should change in https://github.com/dotnet/coreclr/blob/master/src/inc/corprof.idl#L306 as well. corprof.h is regenerated from corprof.idl occasionally so we want to keep the two in sync.

> Do the "WithInfo" functions (e.g., FunctionEnter3WithInfo) also need to change?

I think it would be nice to change all `__stdcall` in corprof.idl/h to STDMETHODCALLTYPE